source/board/nrf52820_microbit_bl.c

```diff
@@ -84,10 +93,44 @@ bool reset_button_pressed()
 
 bool board_bootloader_init()
 {
-    // Lock the bootloader flash, only protects until reset,
-    // so needs to be executed in the bootloader on every startup
-    uint8_t acl_region = 0;
-    NRF_ACL->ACL[acl_region].ADDR = DAPLINK_ROM_BL_START;
-    NRF_ACL->ACL[acl_region].SIZE = DAPLINK_ROM_BL_SIZE;
-    NRF_ACL->ACL[acl_region].PERM = ACL_ACL_PERM_WRITE_Disable << ACL_ACL_PERM_WRITE_Pos;
+    if (NRF_FICR->INFO.PART == 0x52833) {
+        // nRF52833
+        // Configure the NFC pins as GPIO in the UICR if not done already
+        volatile uint32_t* const nrf_uicr_nfcpins = (uint32_t *) NRF52833_UCIR_NFCPINS_ADDRESS;
+        if ((*nrf_uicr_nfcpins & NRF52833_UICR_NFCPINS_PROTECT_Msk) ==
+                (NRF52833_UICR_NFCPINS_PROTECT_NFC << NRF52833_UICR_NFCPINS_PROTECT_Pos)) {
+            nrf_nvmc_mode_set(NRF_NVMC, NRF_NVMC_MODE_WRITE);
+            *nrf_uicr_nfcpins &= ~NRF52833_UICR_NFCPINS_PROTECT_Msk;
+            while (!nrf_nvmc_ready_check(NRF_NVMC));
+            nrf_nvmc_mode_set(NRF_NVMC, NRF_NVMC_MODE_READONLY);
+            // Changes only take effect after a system reset
+            // TODO: This does not seem to reset the system correctly, it hangs somewhere
+            NVIC_SystemReset();
+        }
```

Comment:
Rather than trying to configure the NFC pins, let's apply the flash protection if the NFC pins are not configure as GPIO.

Reply:
Just to clarify, we check if the NFC pins are configured as GPIO:
- Configured as GPIO: Check the pin and apply protection based on pin state
- Configured as NFC: Apply the protection

Reply:
Added https://github.com/mathias-arm/DAPLink/pull/13/commits/7226667d741eb7ba5b62beb346616e4ae33b33be
I'm assuming the existing check for "configured as NFC" is the same as checking for "not configured as GPIO".